### ISSUE TITLE
Community page: “seasoning” and filters

### ISSUE THREAD

**Issue by mkalininait** [rails-girls-summer-of-code/rgsoc-teams#594]
Users reported that it was hard to use the community functionality in the Teams app, because it doesn't allow you to filter users.

It would be useful to provide filters by:
 - availability in the upcoming season
 - location
 - technologies (see [Issue 593](https://github.com/rails-girls-summer-of-code/rgsoc-teams/issues/593))

**One possible way to implement "seasoning":**
 - Add a checkbox "Active in the upcoming season" - should be reset to 0 for all profiles at the end of each season

**Comment by carpodaster** [comment]
It is true that with the Teams App keeping users of past seasons, it's hard to know whether or not someone who once put them on the roster as willing to help would still be up for it in the upcoming season.

I don't know if a filter solves that, since it would require all users on the Teams App to proactively check that option. How would we do that? Emailing all users?

**Comment by mkalininait** [comment]
@carpodaster I think that we can't consider a user active, unless he/she explicitly marks him/herself active in the Teams app. Emailing and tweeting would be the most obvious ways to remind them to set a check.
I guess that this check would be most useful for coaches and students looking for coaches. The "Active in the upcoming season" check wouldn't guarantee availability of a coach of course, but it would increase the probability of it. If a student can't find an active coach, she can always try to contact a coach without the "Active in the upcoming season" check.

**Comment by F3PiX** [comment]
If that is the use case: students finding coaches, how about this: add a filter for NON-active coaches, let coaches be able to adjust their availability in their profile.

Easier for coaches, bigger chance for students to persuade a coach to join their team.

**Comment by mkalininait** [comment]
@F3PiX Do you mean that the coaches who won't be available should set a check "I'm not available in the upcoming season"? I'm afraid that people who can't or lost interest in coaching may not bother reporting this. Or did I misunderstand you?

**Comment by F3PiX** [comment]
Exactly! And they will set that option in their profile as soon as it is important to THEM, ie, the moment a student asks them (or with a ping from you to the coaches in Slack).

**Comment by alicetragedy** [comment]
I agree with Maria: if someone has joined the Teams App once and showed interest in coaching (e.g. in 2014) they might not go back to the Teams app if they've lost interest to say that they will not be coaching anymore and check a box. 
On the other hand, someone interested in coaching will be logging into the teams app again, and would be able to check said box.

I think what we could do once this has been implemented would be to email all users and let them know about this change; that way, if a user from 2014 is interested in coaching again, they might actively log in and select the “available for the next season” box.

**Comment by carpodaster** [comment]
> On the other hand, someone interested in coaching will be logging into the teams app again, and would be able to check said box.

Hm, I question that – that is: _I_ would not be that person. As a coach or mentor you only used the teams app once: when you signed up. They probably have long forgotten about the existence of the app, let alone the URL to get there.

I don't like the idea of emailing all users on the teams app every year anew. Because of that, I am more in favor of @F3PiX's suggestion.

**Comment by mkalininait** [comment]
We could use our blog and twitter to ask users to "sign up" for the next season if they are interested.

Last year we had quite a few emails from people who wanted to be coaches but didn't know what to do to become one. With the "available" checkbox we could ask them to set the check and wait :) (which btw could be quite efficient if we improve the community page by adding availability filter and some other ones)

And of course, the "available" checkbox would make it easier to make an active search. I don't see how the "not available" checkbox would make it. I imagine myself looking for a coach: there are 194 coaches in the database, if 50 people mark themselves "unavailable", I will still have 144 coaches to choose from. If I look for a remote coach, I can't even filter them by location. With the "available" checkbox, on the contrary, if 50 people mark themselves "available", it'd make my choice easier. So it's 144 vs 50, and we should keep in mind that the amount of users in the database will grow and grow with time (and the mentioned difference will grow with it I guess).

Another thing is that we could actively agitate people to support RGSoC and set the "available" checkbox via social media, while with the "unavailable" checkbox if would look weird ("Please set a check 'unavailable' if you don't want to support RGSoC!" :)) So even leaving aside the motivational part, I believe that we will always have more people who want to mark themselves "available" than people who get annoyed by students and want to set the "unavailable" checkbox.

**Comment by F3PiX** [comment]
That are quite a few interesting issues. No I feel even stronger against the availability filter LOL

- Looking for a coach:
If we want to improve the coach searching for the students, I'd rather add a filter on the languages or frameworks they are comfortable with. 
Plus, with the newly added geocoding, we can add a filter on location.
Also, looking for a 'remote' coach will not be easier with availability; maybe add a 'love to coach remotely' checkbox.
Alll together: would plead for a bit more verbose coaches profile.
And then we can ask them upfront how much time they have available! Yay!
That said: I'd say start with the languages filter, rather the availability. Wdyt?

- Aspiring coaches:
When the email their wish to coach, we still can redirect them to the profile pages. 

- Filter output
With all of the above, I'd rather add a filter for the 'last touched' profiles, as an indication for actively engaged people.
When emailing existing coaches/users, or on twitter, you can ask them to complement their profile.
(Ever heard of a dev who didn't change at least one language in a year ;-) )

- And even better:
I would love if we could offer the students more help with selecting coaches. The languages, remote, and location are some options, but I would love if we could ask the coaches about their, let's say, coaching abilities.

**Comment by mkalininait** [comment]
@F3PiX I totally agree that we need more info about our coaches, especially tech info. There is an issue about it: https://github.com/rails-girls-summer-of-code/rgsoc-teams/issues/593 (see the last suggestion there). Comments are welcome!
And I agree that it would be cool to filter by location and languages/frameworks, and maybe something else. As well as by availability 😄  😉

**Comment by klappradla** [comment]
Moving this into the [Community Page(s)](https://github.com/rails-girls-summer-of-code/rgsoc-teams/milestone/13) milestone, the next project @juuh42dias and @branquinhoaa will tackle 💪 

Let's take this discussion as a starting point to come up with the necessary tickets for this feature. Once the smaller sized tickets are created, we can close this issue right here.

Summing up the so far "decided" upon steps here:
- enhance coaches' profile information (info about languages, availability, etc.)
- come up with indicator for coaches' profiles to signalize _"availability as a coach for the upcoming season"_
  - behavior: the `availability` is set to "unavailable" again for each season and has to be updated manually by the users themselves
- enhance the community page listing all users:
  - better general overviw
  - better filter / search possibilities: filter by availability for the season, by location, by availablity in terms of time, by language / fields of knowledge
- mechanism to have users constantly return to the app and keep their profile updated:
  - none (use other channels to remind people to keep their profile up to date)
  - alternative: one reminder email per pre-season

**Comment by branquinhoaa** [comment]
We want to start it right now!
Before that, I have some questions:

- Enhance coach's profile information belongs to another milestone (Application Process 2017). Will the issue #593 be included in the community milestone?  @mkalininait 

- We are not sure if there was a decision regarding the way to search by coach's availability (apparently you guys are divided :joy: ) 

- Are we allowed to create the issues related to @klappradla suggestions?

**Comment by alicetragedy** [comment]
hey @branquinhoaa, yeah I think we are still divided on that and no real decision has been made.. 🙈

**Comment by klappradla** [comment]
Ok, so the question which channel (email vs. social media vs. whatever) should be used to ping the coaches again to update their profile and availability can be postponed in my opinion - not super necessary to have this in place right away.

For the open question
> check for available vs. unavailable

I would say, there's no huge difference on the technical side. So if we start with **the available approach™** and find out later that "unavailable" would have been a better idea, I'm pretty sure this can be changed within a few simple commits.

Therefore my suggestion to not block @branquinhoaa and @juuh42dias: let's go for **the available approach™** for now. It appears slightly easier to understand and all the wording around it a bit more "obvious" then in the other case - therefore I'd consider it the easier approach and good enough for a beginning.

**Comment by branquinhoaa** [comment]
Are all the users allowed to set her availability or just the coaches?

**Comment by carpodaster** [comment]
@branquinhoaa I would say all users, since a user only becomes a coach when they have been assigned that role in a team in a year. Showing interest in something (coach, supervisor, …) should be regardless of whether or not one has previously assumed that role in RGSoC.

**Comment by mkalininait** [comment]
@branquinhoaa 

>Enhance coach's profile information belongs to another milestone (Application Process 2017). Will the issue #593 be included in the community milestone? 

The tasks related to enhancing coach's profile were actually moved here: #609, #612. We still want to discuss them probably. I'm not sure who created the community milestone; whoever did it may consider adding #609 and #612 there :)
Regarding #593: I'd actually like to close it.

>Are all the users allowed to set her availability or just the coaches?

I think it makes sense only for coaches.

**Comment by mkalininait** [comment]
@carpodaster 

>since a user only becomes a coach when they have been assigned that role in a team in a year

True, but a user can say that they are interested in "Helping as a coach" in their profile. And this is how we find them on the community page when we filter it by the role = "Coach". 
So theoretically we could assign some fields to users who are interested in being coaches I guess.

**Comment by carpodaster** [comment]
@mkalininait agreed, I buy that 😄

**Comment by branquinhoaa** [comment]
@klappradla in the 

> enhance the community page listing all users:
> better general overview
> better filter/search possibilities: filter by availability for the season, by location, by availablity in terms of time, by language/fields of knowledge

1.  What do you mean by better general overview? Do you want us to change the way user search? like the layout?

2. It is not possible to filter user by language/fields of knowledge because the user does not have this information linked in her profile. The project has tags that we could get if the user was linked to one project, but as @mkalininait has explained in a past issue, the user is linked with all projects that she applied for. With the current user model we don't have access to this information.
